### ISSUE TITLE
Azure App Service Deployment

### ISSUE THREAD

**Issue by faysalfdt** [abpframework/abp#17588]
### Is there an existing article or article request for this?

- [X] I have searched the existing resources

### Describe the article you'd like

I've tried setting up my local docker deployment on azure app service and would really like to see a guided article. I've been having issues with the docker-compose file and seeing the correct deployment configs would really help

**Comment by EngincanV** [comment]
Hi, we have created documentation for Azure app service deployment 👇 
https://docs.abp.io/en/commercial/7.4/startup-templates/application/azure-deployment/azure-deployment?UI=MVC&DB=EF&Tiered=No

**Comment by sturlath** [comment]
I need to take a look at that documentation but you could also view this one https://sturla.io/deploying-blazor-server-to-azure-with-yml-and-bicep-through-github-actions/

**Comment by faysalfdt** [comment]
> Hi, we have created documentation for Azure app service deployment 👇 https://docs.abp.io/en/commercial/7.4/startup-templates/application/azure-deployment/azure-deployment?UI=MVC&DB=EF&Tiered=No

So I followed all of the steps, however my API Host isn't deploying properly when using the github action, when I navigate to the webapp I deployed, it doesn't seem to trigger the dotnet run portion for the API. 

The wwwroot folder has the release binaries, however when I look at the processes inside the VM it doesn't show dotnet running
[logs.zip](https://github.com/abpframework/abp/files/12564447/logs.zip)

**Comment by faysalfdt** [comment]
I think I found a mistake when it's set to Angular/Mongo

![image](https://github.com/abpframework/abp/assets/137331400/e9b99820-f15b-4394-ae8d-0edfd4cf9b5f)



The doc doesn't specify a package, which I think makes it run into an error

This is from the Azure doc for the v2

![image](https://github.com/abpframework/abp/assets/137331400/4cb67c0b-38ac-4b76-ba5f-76a0bd1975d3)

**Comment by skoc10** [comment]
> I think I found a mistake when it's set to Angular/Mongo
> 
> ![image](https://user-images.githubusercontent.com/137331400/266756971-e9b99820-f15b-4394-ae8d-0edfd4cf9b5f.png)
> 
> The doc doesn't specify a package, which I think makes it run into an error
> 
> This is from the Azure doc for the v2
> 
> ![image](https://user-images.githubusercontent.com/137331400/266757014-4cb67c0b-38ac-4b76-ba5f-76a0bd1975d3.png)


Firstly, in this GitHub Actions file, you need to replace the app name and publish profile secret with your own

- The published apihost is then saved as an artifact named .net-apihost to be used in subsequent jobs.
- The .net-apihost artifact, which was saved in the previous job, is downloaded to the runner.
- Using the Azure webapps-deploy action, the apihost is deployed to Azure Web App.

**Comment by faysalfdt** [comment]
I was able to confirm that if you don't specify the package, it will not deploy successfully. 

![image](https://github.com/abpframework/abp/assets/137331400/aa9bdcaf-6e27-429c-8f9f-3b1aaee4a70f)

It's missing in y'alls documentation:
![image](https://github.com/abpframework/abp/assets/137331400/b37a93f6-1f62-4b92-8f4a-596456a48165)

**Comment by skoc10** [comment]
Yes, you are right. The 'package: ./apihost' line was missing for the 'Deploy apihost' step. I've promptly corrected it in the document. Thank you for your contribution.

https://docs.abp.io/en/commercial/7.4/startup-templates/application/azure-deployment/step3-deployment-github-action?UI=NG&DB=EF&Tiered=No

**Comment by faysalfdt** [comment]
Of course! I also wondered if there was any additional steps?

I've hit the following issues/behaviours

When I attempt to login I get the following issues:

Before I hit the login button, I do get a 404 on the getconfigenv rest call
<img width="1705" alt="Screenshot 2023-09-12 at 11 06 01 AM" src="https://github.com/abpframework/abp/assets/137331400/380e62c3-b037-4a1e-9efd-54f9591422ba">

I also get another issue after the login button gets clicked, I get an error 500 and a 302

<img width="1721" alt="Screenshot 2023-09-12 at 11 06 17 AM" src="https://github.com/abpframework/abp/assets/137331400/c3e31cb0-0baa-4f1f-acd2-22559dc30a4c">

**Comment by faysalfdt** [comment]
I'd also like to recommend that there be a additional fix added in the docs for the following when deploying the angular piece to azure static web app

![image](https://github.com/abpframework/abp/assets/137331400/425182d2-e7df-4f21-9358-09a675218bcd)

I had to create a file called staticwebapp.config.json with the following content to fix it.

`{
    "platform": {
      "apiRuntime": "node:18"
    }
}
`
